[Kevin Quek] iP

kev-intq opened the pull request:
# **DukePro** 

> "Stop making fake quotes and crediting it to me" - Sun Tzu([source](https://ifunny.co/picture/stop-making-fake-quotes-up-and-crediting-it-to-me-PoQLZnqk8))

Ever missed an important assignment because it slipped your mind? Worry no more!
DukePro is a simple chatbot aimed at managing your tasks to ensure that you'll never miss another important task ever again. 

## **Its has:** 

- A simple based input method suited for fast typers
- Easy to learn commands
- A quirky assistant bot brimming with personality 
<img src="https://64.media.tumblr.com/e2358326f171b07a12cc09f46adc8a4f/d900fba9b3ad6ca6-88/s400x600/2568b87b5ccf0c7f2440e54df642328b90d7e0f6.png" alt="" width="100"/>

## **All you need to do is**

1.  Download it from [here](https://github.com/kev-intq/ip/releases/tag/Level-10)
2.  Double-click it
3. Add your tasks
4. Let it manage your tasks for you :smiley:

And it is **FREE!**

## Features

- [x] Managing simple tasks
- [x] Managing deadlines and events (with dates)
- [x] Genshin-themed UI

![](https://static.wikia.nocookie.net/gensin-impact/images/7/74/Icon_Emoji_066_Hu_Tao_Reciting_poetry.png/revision/latest/scale-to-width-down/250?cb=20210906044059)

![07 02 2022_03 06 34_REC](https://user-images.githubusercontent.com/77195251/152698580-0cc47ebc-f403-400a-af76-698a96441d14.png)

If you are familiar with CSS and JavaFX, you can further customize the UI to suit your needs. Here is a snippet of the GUI code:
```java
private DialogBox(String text, Image img, String user) {
        try {
            FXMLLoader fxmlLoader;
            if (user.matches("user")) {
                fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBoxUser.fxml"));
            } else {
                fxmlLoader = new FXMLLoader(MainWindow.class.getResource("/view/DialogBoxBot.fxml"));
            }
            fxmlLoader.setController(this);
            fxmlLoader.setRoot(this);
            fxmlLoader.load();
        } catch (IOException e) {
            e.printStackTrace();
        }

        dialog.setText(text);
        displayPicture.setFill(new ImagePattern(img));
    }
```

Commit:
Add Gradle support

kev-intq committed:
Level-1

kev-intq committed:
Level-2

kev-intq committed:
Level-3

kev-intq committed:
Level-4

kev-intq committed:
Level-4

kev-intq committed:
Level-4 Final

kev-intq committed:
A-TextUiTesting

kev-intq committed:
Level-5

kev-intq committed:
Level-5.1

kev-intq committed:
Level-6

kev-intq committed:
Refactor code for more OOP

kev-intq committed:
Refactor code for more OOP


Refactor code

kev-intq committed:
Revert "Merge remote-tracking branch 'origin/experimental-parser' into experimental-parser"

This reverts commit c7d17fa91645d47bf862786588794f2df6854cfc.

kev-intq committed:
Merge remote-tracking branch 'origin/experimental-parser' into experimental-parser

kev-intq committed:
Merge branch 'experimental-parser'

kev-intq committed:
Cleaned up code


Cleaned up code

kev-intq committed:
Add save and load functionality

kev-intq committed:
Add date and time support

kev-intq committed:
Merge branch 'branch-Level-7'

Added save and load functionality

kev-intq committed:
Merge branch 'branch-Level-8'

kev-intq committed:
Add command classes and refactor existing classes

kev-intq committed:
Arrange classes into packages

Commands and CommandResults classes moved to com.duke.command package.

DukeException classes moved to com.duke.exception package.

Task classes moved to com.duke.tasks package.

Parser, Ui, Tasklist, Storage moved to com.duke.modules package,

kev-intq committed:
Add JUnit tests

Added Junit tests for Parser and Tasklist

kev-intq committed:
Update gitignore

Added the following file to .gitignore:

-MANIFEST.MF

kev-intq committed:
Add documentation to project

Java documentation has been added to all classes other than Tasks (and its subclasses)

kev-intq committed:
Reformat code to fit coding convention

kev-intq committed:
Implement find keyword functionality

Implemented functionality to find tasks in the task list containing specified keywords by the user.

kev-intq committed:
Implement find keyword functionality

Implemented functionality to find tasks in the task list containing specified keywords by the user.

kev-intq committed:
Merge branch 'branch-Level-9'

kev-intq committed:
Merge branch 'branch-A-JavaDoc'

kev-intq committed:
Update classes with Java Documentation

Updated TaskList, CommandFind with Java Documentation on added methods.

kev-intq committed:
Merge branch 'branch-A-CodingStandard'

kev-intq committed:
Merge remote-tracking branch 'origin/branch-Level-9' into branch-Level-9

kev-intq committed:
Merge branch 'branch-Level-9'

Attempt to fix merge of branch-Level-9 into master branch due to incomplete progress tracked in iP dashboard.

kev-intq committed:
Merge remote-tracking branch 'origin/add-gradle-support'

kev-intq committed:
Add Gradle support

Added Gradle support for project.
Added more JUnit tests for commands and Duke module classes.

kev-intq committed:
Add checkstyle to project and fix coding style violations

hqhqhq1 commented on src/main/java/com/duke/modules/Parser.java:
```diff
@@ -0,0 +1,70 @@
+     * @param string String of user input.
+     * @return A command result from the execution of the command.
+     */
+    public CommandResult parse(String string) {
```
very neat and clean 👍

hqhqhq1 commented on src/main/java/com/duke/modules/Parser.java:
```diff
@@ -0,0 +1,70 @@
+        case "":
+            cmdResult = new CommandResult("*crickets*");
+            break;
+        case "bye":
```
does it account for uppercase letters for example if I input "Bye" what will happen

kev-intq replied:
Thank you for pointing this out. 😄

hqhqhq1 commented on src/main/java/com/duke/modules/Storage.java:
```diff
@@ -0,0 +1,121 @@
+     * @throws IOException On failure of read or write operations.
+     */
+    public boolean saveList(ArrayList<Task> taskList) throws IOException {
+        boolean success = false;
```
boolean var can be declared as isSuccess

hqhqhq1 commented on src/main/java/com/duke/tasks/Task.java:
```diff
@@ -0,0 +1,46 @@
+package com.duke.tasks;
+
+public class Task {
+    protected boolean status;
```
maybe can declare as isDone

leeyiheng12 commented on src/main/java/com/duke/modules/Storage.java:
```diff
@@ -0,0 +1,121 @@
+public class Storage {
+    private static Storage STORAGE_INSTANCE;
+    private String directoryPath = System.getProperty("user.dir") + "/data/";
+    private String listFile = "listData.txt";
```
if planning to keep the name of the file constant, can consider using uppercase for listFile

leeyiheng12 commented on src/main/java/com/duke/command/CommandAdd.java:
```diff
@@ -0,0 +1,134 @@
+            }
+            taskType = Task.TaskType.TODO;
+        } else if (type.equals("deadline")) {
+            if (!args.matches("\\S+.*\\s/by\\s\\S+.*")) {
```
nice use of regex!

leeyiheng12 commented on src/main/java/com/duke/command/CommandAdd.java:
```diff
@@ -0,0 +1,134 @@
+     */
+    private String taskAdder(String type, String args)
+            throws DukeInvalidArgumentException, DateTimeParseException, IOException {
+        String message = "";
```
unused variable?

leeyiheng12 commented on src/main/java/com/duke/command/CommandMark.java:
```diff
@@ -0,0 +1,77 @@
+                    + Storage.getInstance().getDirectoryPath());
+        }
+    }
+
```
thorough checking of different possible exceptions, each returning different prompts, is a nice touch 👍

leeyiheng12 reviewed (commented):
LGTM!! A lot of functionality of the bot encapsulated in relevant classes 👍

kev-intq committed:
Add GUI to Duke

Added GUI support to Duke.
Refactored Duke to take in user input from GUI.

kev-intq committed:
Clean up coding violations and unused variables

kev-intq committed:
Fix errors due to renamed variables

kev-intq committed:
Update README.md

kev-intq committed:
Add Java Assertions and bug fixes

Added assert checks to private methods in class CommandAdd.

This ensures type of task (Tiodo, Deadline or Event) is specified in the methods.

Bug fixes:
* Removed smiley face character in messages containing it. Avoids incompatibility issues with different device formats.

kev-intq committed:
Improve code quality

Removed redundant methods which were unused through the lifetime of the program.

Changed access modifier for taskMarker to private, since it is only used within the CommandMark class itself.

kev-intq committed:
Merge pull request #1 from kev-intq/branch-A-Assertions

Add Java Assertions and bug fixes

kev-intq committed:
Merge pull request #2 from kev-intq/branch-A-CodeQuality

Improve code quality

kev-intq committed:
Set up Continuous Integration

Used GitHub Actions to set up Continuous Integration (CI).

Removed unused imports in class Duke.

Fixed assertion tests related to class CommandAdd.

kev-intq committed:
Remove IO Redirection tests from workflow

kavxya commented on src/main/java/com/duke/tasks/Todo.java:
```diff
@@ -9,7 +9,7 @@ public Todo(String description) {
     public String getSaveDescription() {
         return String.format("%s | %s | %s ",
-                getClass().getName(), status == true ? 1 : 0, description);
+                getClass().getSimpleName(), status == true ? 1 : 0, description);
```
I like that getSimpleName goes into more detail of the name, just curious is there a more detailed name you wish to return as well?

kev-intq replied:
The getName() method worked for me fine until I packaged my classes, which following that it would give me:

"com.duke.task.Todo" 

instead of 

"Todo" 

So I had to change it into getSimpleName() instead,

kev-intq committed:
Expand GUI for extension

Included tab support in GUI in preparation for BCD-Extensions.

kev-intq committed:
Update README.md

kev-intq committed:
Add C-Help Extension

Added help menu extension with tab support.

kev-intq committed:
Merge branch 'branch-BCD-Extension'

kev-intq committed:
Fix typo causing jar errors

Corrected case-sensitive filepaths. This is done as IDE is case-insensitive when dealing with filepaths, but built-jar is not.

Added extra bot reaction image.

Adjusted insets for dialog boxes.

kev-intq committed:
Fix unused imports

kev-intq committed:
Fixed tests

kev-intq committed:
Fix boolean names

Changed boolean names to be more in accordance to Java coding standard.

kev-intq committed:
Update README.md

kev-intq committed:
Update README.md

kev-intq committed:
Update README.md

kev-intq committed:
Update README.md

kev-intq committed:
Update README.md
Update README.md
Update README.md
Update README.md

kev-intq committed:
Finalize iP

Remove redundant strings from Ui class.
Added welcome message.

kev-intq committed:
Merge remote-tracking branch 'origin/master'

kev-intq committed:
Update README.md

kev-intq committed:
Update gradle build file

Added in explicit JavaFX dependencies.

kev-intq committed:
Merge remote-tracking branch 'origin/master'
